### PR TITLE
Add: specify that reason is a field to record the reason why failed

### LINKED ISSUE #103737: scheduler: panic in AppendReason, when status is nil (= success)
#### What happened:

AppendReason panic when status is nil. 
https://github.com/kubernetes/kubernetes/blob/818ed1afff306f8b310c56083e0c6be4fa93ce8a/pkg/scheduler/framework/interface.go#L185

Some plugins return nil to mean success. 

(Or, shouldn't Status.reason be used on success status? I use Status.reason field as a reason for why it is success.)

#### What you expected to happen:

not panic or describe this behaviour on the document

#### How to reproduce it (as minimally and precisely as possible):

https://play.golang.org/p/WWohPs5qYS5

#### Anything else we need to know?:

### PR DESCRIPTION
**Pull request by sanposhiho**
Hi team :)

#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

Add documentation that tells `reason` field is only used when it is not success status.

#### Which issue(s) this PR fixes:

Fixes #103737

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/interface.go [hunk 1]
```diff
@@ -135,7 +135,8 @@ func NewPodsToActivate() *PodsToActivate {
 // code is not `Success`, the reasons should explain why.
 // NOTE: A nil Status is also considered as Success.
 type Status struct {
-	code    Code
+	code Code
+	// reasons is field that records the reason why it is not success.
```
Comment (Huang-Wei) [inline pkg/scheduler/framework/interface.go:139]: Hmm... I don't think it's that needed - L135 already explains it.

Comment (sanposhiho) [reply pkg/scheduler/framework/interface.go:139]: IMO, L135 says that if code is non-success, you should explain why it is not successful on reasons field. 
But it doesn't mean that reasons field is only for failed reasons (not for success reasons). 

This is just my opinion, so please feel free to close this PR if you don't think it's necessary :)

Comment (alculquicondor) [reply pkg/scheduler/framework/interface.go:139]: The way you are phrasing this also doesn't imply that reasons shouldn't be used for success.

I would modify L133-136 to add: When code is Success, all the other fields should be empty.

Comment (sanposhiho) [reply pkg/scheduler/framework/interface.go]: Fixed, thanks @alculquicondor :)

**Comment by alculquicondor** [comment]
/approve
/lgtm

Thanks. Note that we are under codefreeze, so the change wouldn't be merged until sometime next month.

**Pull request merged**